File: ui/splash.py
from rich.console import Console

from .theme import Theme


def show_splash():
    """
    Displays the styled ASCII splash screen with correct alignment using Theme colors.
    """
    console = Console()

    # Geometric ASCII Art
    logo = [
        r"███╗   ███╗  ██████╗  ██████╗  ██████╗  ██╗  ██╗ ██╗   ██╗",
        r"████╗ ████║ ██╔═══██╗ ██╔══██╗ ██╔══██╗ ██║  ██║ ╚██╗ ██╔╝",
        r"██╔████╔██║ ██║   ██║ ██████╔╝ ██████╔╝ ███████║  ╚████╔╝ ",
        r"██║╚██╔╝██║ ██║   ██║ ██╔══██╗ ██╔═══╝  ██╔══██║   ╚██╔╝  ",
        r"██║ ╚═╝ ██║ ╚██████╔╝ ██║  ██║ ██║      ██║  ██║    ██║   ",
        r"╚═╝     ╚═╝  ╚═════╝  ╚═╝  ╚═╝ ╚═╝      ╚═╝  ╚═╝    ╚═╝   ",
    ]

    # --- Dimensions ---
    content_width = len(logo[0])
    padding = 4
    inner_width = content_width + (padding * 2)

    # --- Text Content ---
    left_text = "simple file converter"
    right_text = "made by kazuto"
    spacer_len = inner_width - 8 - len(left_text) - len(right_text)

    # --- Render ---
    # Define colors from Theme
    border_c = Theme.BORDER
    logo_c = Theme.HEADER
    text_c = Theme.TEXT

    # Top Border
    console.print(f"[{border_c}]╔{'═' * inner_width}╗[/{border_c}]")

    # Top Spacer
    console.print(f"[{border_c}]║{' ' * inner_width}║[/{border_c}]")

    # Logo Section
    for line in logo:
        console.print(
            f"[{border_c}]║[/{border_c}]{' ' * padding}[{logo_c}]{line}[/{logo_c}]{' ' * padding}[{border_c}]║[/{border_c}]"
        )

    # Middle Spacer
    console.print(f"[{border_c}]║{' ' * inner_width}║[/{border_c}]")

    # Subtext Section
    console.print(
        f"[{border_c}]║[/{border_c}]    [{text_c}]{left_text}{' ' * spacer_len}{right_text}[/{text_c}]    [{border_c}]║[/{border_c}]"
    )

    # Bottom Spacer
    console.print(f"[{border_c}]║{' ' * inner_width}║[/{border_c}]")

    # Bottom Border
    console.print(f"[{border_c}]╚{'═' * inner_width}╝[/{border_c}]")

File: src/ui/splash.rs
/// splash.rs - displays the startup ascii splash screen

use console::style;
use crate::ui::theme::Theme;

/// show the styled ascii splash screen with correct alignment
pub fn show_splash() {
    // geometric ascii art
    let logo = [
        r"███╗   ███╗  ██████╗  ██████╗  ██████╗  ██╗  ██╗ ██╗   ██╗",
        r"████╗ ████║ ██╔═══██╗ ██╔══██╗ ██╔══██╗ ██║  ██║ ╚██╗ ██╔╝",
        r"██╔████╔██║ ██║   ██║ ██████╔╝ ██████╔╝ ███████║  ╚████╔╝ ",
        r"██║╚██╔╝██║ ██║   ██║ ██╔══██╗ ██╔═══╝  ██╔══██║   ╚██╔╝  ",
        r"██║ ╚═╝ ██║ ╚██████╔╝ ██║  ██║ ██║      ██║  ██║    ██║   ",
        r"╚═╝     ╚═╝  ╚═════╝  ╚═╝  ╚═╝ ╚═╝      ╚═╝  ╚═╝    ╚═╝   ",
    ];

    // --- dimensions ---
    let content_width = logo[0].chars().count();
    let padding = 4;
    let inner_width = content_width + (padding * 2);

    // --- text content ---
    let left_text = "simple file converter";
    let right_text = "made by kazuto";
    let spacer_len = inner_width.saturating_sub(8 + left_text.len() + right_text.len());

    // --- render ---

    // top border
    println!(
        "{}",
        style(format!("╔{}╗", "═".repeat(inner_width))).fg(Theme::BORDER)
    );

    // top spacer
    println!(
        "{}",
        style(format!("║{}║", " ".repeat(inner_width))).fg(Theme::BORDER)
    );

    // logo section
    for line in &logo {
        let left_border = style("║").fg(Theme::BORDER);
        let right_border = style("║").fg(Theme::BORDER);
        let logo_text = style(*line).fg(Theme::HEADER);
        let pad = " ".repeat(padding);

        println!("{left_border}{pad}{logo_text}{pad}{right_border}");
    }

    // middle spacer
    println!(
        "{}",
        style(format!("║{}║", " ".repeat(inner_width))).fg(Theme::BORDER)
    );

    // subtext section
    let left_b = style("║").fg(Theme::BORDER);
    let right_b = style("║").fg(Theme::BORDER);
    let sub = style(format!(
        "{left_text}{}{right_text}",
        " ".repeat(spacer_len)
    ))
    .fg(Theme::TEXT);

    println!("{left_b}    {sub}    {right_b}");

    // bottom spacer
    println!(
        "{}",
        style(format!("║{}║", " ".repeat(inner_width))).fg(Theme::BORDER)
    );

    // bottom border
    println!(
        "{}",
        style(format!("╚{}╝", "═".repeat(inner_width))).fg(Theme::BORDER)
    );
}
